src/Backend/Modules/MediaLibrary/Engine/CacheBuilder.php

```diff
@@ -0,0 +1,275 @@
+<?php
+
+namespace Backend\Modules\MediaLibrary\Engine;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Backend\Modules\MediaLibrary\Domain\MediaGroup\MediaGroup;
+use Backend\Modules\MediaLibrary\Domain\MediaItem\MediaItem;
+use Backend\Modules\MediaLibrary\Domain\MediaGroup\MediaGroupRepository;
+
+/**
+ * In this file we store all generic functions that we will be using in the Media module
+ */
+class CacheBuilder
+{
+    /**
+     * @var CacheItemPoolInterface
+     */
+    protected $cache;
+
+    /**
+     * @var \SpoonDatabase
```

Comment:
you are using SpoonDatabase here, can't you swap it out for a repositry that interacts with the database via doctrine?